jdl/converters/json-to-jdl-converter.ts

```diff
@@ -116,12 +116,17 @@ function cleanYoRcFileContent(yoRcFileContent: Record<string, any>) {
   return yoRcFileContent;
 }
 
-function getJSONEntityFiles(applicationDirectory: string) {
+function getJSONEntityFiles(applicationDirectory: string, yoRcFileContent: Partial<JSONRootObject>): Map<string, JSONEntity> {
   const entities: Map<string, JSONEntity> = new Map();
   fs.readdirSync(path.join(applicationDirectory, '.jhipster')).forEach(file => {
+    const entityName = file.slice(0, file.indexOf('.json'));
+    if (!yoRcFileContent.entities?.includes(entityName)) {
+      throw new Error(
+        `Mismatch identified between the application description amount of entity listed and the entity found in the .jhipster folder for entity name: ${entityName}`,
+      );
+    }
```

Comment:
JSON entities field is not required to be in sync.
The reason for it is to import jdls.
Import jdl operation is done in memory, having entities field avoids having to glob the mem-fs files to discover entities.

Reply:
Sorry to ask stupid question: Is this `json-to-jdl.convert()`method used in the import-jdl process?
This exception is technically useful to ensure that the JSONx typescript object is inline with the .yo-rc json content. I would keep it if feasible, otherwise will drop

Reply:
json-to-jdl is used in export-jdl.